ast/desugar/Desugar.cc

```diff
@@ -2197,6 +2197,22 @@ ExpressionPtr node2TreeImpl(DesugarContext dctx, unique_ptr<parser::Node> what)
                 result = std::move(res);
             },
             [&](parser::Case *case_) {
+                if (dctx.preserveConcreteSyntax) {
+                    Send::ARGS_store args;
+                    args.emplace_back(node2TreeImpl(dctx, std::move(case_->condition)));
+                    for (auto it = case_->whens.begin(); it != case_->whens.end(); ++it) {
+                        auto when = parser::cast_node<parser::When>(it->get());
+                        ENFORCE(when != nullptr, "case without a when?");
```

Comment:
[super nit]
Would it make sense to rephrase the message in `ENFORCE`? It could be something like 
`found a case without when in desugaring stage for extract variable code action`

Reply:
I copied this `ENFORCE` from the regular/non-extract variable case, and if it occurs here, it should occur there too, so I don't think it's worth mentioning the message that's here (and the enforce failure will mention the line number if we do need to know that).